clients/src/test/java/org/apache/kafka/clients/MetadataTest.java

```diff
@@ -76,6 +76,67 @@ public void testMetadata() throws Exception {
         assertTrue("Update needed due to stale metadata.", metadata.timeToNextUpdate(time) == 0);
     }
 
+    private static void checkTimeToNextUpdate(long refreshBackoffMs, long metadataExpireMs) {
+        Metadata metadata = new Metadata(refreshBackoffMs, metadataExpireMs);
+        long now = 10000;
+
+        // Metadata timeToNextUpdate is implicitly relying on the premise that the currentTimeMillis is always
+        // larger than the metadataExpireMs or refreshBackoffMs.
+        // It won't be a problem practically since all usages of Metadata calls first update() immediately after
+        // it's construction.
+        assert now > metadataExpireMs;
+        assert now > refreshBackoffMs;
```

Comment:
Because this isn't testing Metadata's validity. This is for validating test's validity.
As the above comment explains, we're expecting that the current time in milliseconds won't be smaller than `refreshBackoffMs` or `metadataExpireMs` and in fact this test fails when 10001 is given for `metadataExpireMs` for example. However that case is meaningless so shouldn't be tested but if we use junit's assert here that would be misleading for developers I guess?
Then what would be best choice here?
- assert
- junit's assert
- remove this check and just leave a comment